src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCacheCursor.cs

```diff
@@ -24,16 +24,16 @@ public class SimpleQueueCacheCursor : IQueueCacheCursor
 
         internal bool IsSet => Element != null;
 
-        internal void Reset(StreamSequenceToken token)
+        internal void Set(LinkedListNode<SimpleQueueCacheItem> item)
         {
-            Element = null;
-            SequenceToken = token;
+            Element = item;
+            SequenceToken = item?.Value.SequenceToken;
```

Comment:
why would item ever be null? Its better to have the contract more narrow (set on non null value, reset on null value) instead of not well specified interface.
If you are unsure, add NullArgException.